src/backend/distributed/operations/worker_split_copy_udf.c

```diff
@@ -80,9 +80,18 @@ worker_split_copy(PG_FUNCTION_ARGS)
 	}
 
 	EState *executor = CreateExecutorState();
-	DestReceiver *splitCopyDestReceiver = CreatePartitionedSplitCopyDestReceiver(executor,
-																				 shardIntervalToSplitCopy,
-																				 splitCopyInfoList);
+	DestReceiver *destReceiver = NULL;
+	if (list_length(splitCopyInfoList) == 1)
+	{
+		destReceiver = *CreateShardCopyDestReceivers(executor, shardIntervalToSplitCopy,
```

Comment:
can we call CreateShardCopyDestReceiver directly? this seems a bit hacky

Reply:
ShardMove calling 'worker_split_copy_udf' is not that clean as it is tied to Split semantics. Why not introduce a new and simpler worker_shard_copy_udf?